internal/util/features.go

```diff
@@ -0,0 +1,70 @@
+/*
```

Comment:
:memo: I felt that `util` was a reasonable location for this functionality in our current architecture. I also considered creating a separate `feature` package. I ultimately settled on using `util` because it seemed like overkill to move this to a separate package, but I'm certainly up for thoughts/recommendations.